node.gyp

```diff
@@ -177,7 +190,83 @@
   'targets': [
     {
       'target_name': '<(node_core_target_name)',
-      'type': '<(node_target_type)',
+      'type': 'executable',
+      'dependencies': [
+        '<(node_lib_target_name)'
+      ],
+     'sources': [
+        'src/node_main.cc'
+      ],
+      'include_dirs': [
+        'src',
+        'deps/v8/include',
+      ],
+      'conditions': [
+        [ 'node_intermediate_lib_type=="static_library" and '
+            'node_shared=="true" and OS=="aix"', {
+           # For AIX, shared lib is linked by static lib and .exp. In the
+           # case here, the executable needs to link to shared lib.
+           # Therefore, use 'node_aix_shared' target to generate the
+           # shared lib and then executable.
+           'dependencies!': [ '<(node_lib_target_name)' ],
+           'dependencies': [ 'node_aix_shared' ],
+        }],
+        [ 'node_intermediate_lib_type=="static_library" and '
+            'node_shared=="false"', {
+          'includes': [
+            'node.gypi'
+          ],
+          'xcode_settings': {
+            'OTHER_LDFLAGS': [
+              '-Wl,-force_load,<(PRODUCT_DIR)/<(STATIC_LIB_PREFIX)'
+                  '<(node_core_target_name)<(STATIC_LIB_SUFFIX)',
+            ],
+          },
+          'msvs_settings': {
+            'VCLinkerTool': {
+              'AdditionalOptions': [
+                '/WHOLEARCHIVE:<(PRODUCT_DIR)\\lib\\'
```

Comment:
Backward slashes aren't wrong but forward slashes should work too and don't need escaping.

Reply:
it's a bug or a change in vs2017. forward slash (`/`) doesn't work for `/WHOLEARCHIVE` anymore. need to use backward slash (`\`) here.

Reply:
Can you add a explaining comment?  Thanks.